src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,42 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.health = health;
+    this.name = name;
+
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name, health);
+
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animalBited) {
+    if (animalBited instanceof Herbivore && animalBited.hidden === false) {
+      animalBited.health -= 50;
+
+      if (animalBited.health <= 0) {
+        const filtredAnimals = Animal.alive.filter(
+          (animal) => animal !== animalBited,
+        );
+
+        Animal.alive = filtredAnimals;
+      }
```

Comment:
The logic for removing a dead animal from the 'Animal.alive' array is correct, but the variable name 'filtredAnimals' contains a typo. It should be 'filteredAnimals' for proper English spelling.